pkg/httpd/chat.go

```diff
@@ -90,15 +90,22 @@ func (h *chat) chat(
 		chatGPTParam.Proxy,
 	)
 
+	startIndex := len(conversation.HistoryRecords) - 15
+	if startIndex < 0 {
+		startIndex = 0
+	}
+	contents := conversation.HistoryRecords[startIndex:]
+
 	openAIConn := &srvconn.OpenAIConn{
 		Id:          conversation.Id,
 		Client:      c,
 		Prompt:      chatGPTParam.Prompt,
 		Model:       chatGPTParam.Model,
-		Contents:    conversation.HistoryRecords,
+		Contents:    contents,
 		IsReasoning: false,
 		AnswerCh:    answerCh,
 		DoneCh:      doneCh,
+		Type:        h.termConf.ChatAIType,
 	}
 
 	go openAIConn.Chat(&conversation.InterruptCurrentChat)
```

Comment:
There is no need to perform any differences checks since this function appears incomplete. Additionally, there seems to be an issue with assigning values outside of its scope which could lead to unexpected behavior when accessing `content`. Please review all lines carefully before proceeding further.

To optimize efficiency of chat AI operations:

1. Implement error handling for connection pool failure or timeout issues. This can help manage resources efficiently.

2. Refactor the content assignment logic into a separate method within `OpenAIConnect`, where you should use reflection for dynamic types instead of hard-coded types (`&srvconn.OpenAIConn`).

3. Consider adding logging for debugging purposes. This would allow you to monitor interactions better in the case of bugs.